File: src/main/java/com/cloudogu/gitops/config/scm/util/GitlabConfig.java
package com.cloudogu.gitops.config.scm.util;

import com.cloudogu.gitops.config.Credentials;

public interface GitlabConfig {
  String getUrl();

  String getParentGroupId();

  String getDefaultVisibility();

  String getGitOpsUsername();

  Credentials getCredentials();
}

File: src/main/kotlin/com/cloudogu/gitops/config/scm/util/GitlabConfig.kt
package com.cloudogu.gitops.config.scm.util

import com.cloudogu.gitops.config.Credentials

interface GitlabConfig {
    val url: String?
    val parentGroupId: String?
    val defaultVisibility: String?
    val gitOpsUsername: String?
    val credentials: Credentials?
}
